src/hooks/useDetectDevice.ts

```diff
@@ -0,0 +1,76 @@
+import { useState, useEffect } from "react";
+
+export interface DeviceState {
+  isSupported: boolean;
+  deviceType: "Mobile" | "Tablet" | "Desktop";

```

Comment:
Could you extract this to an exported type please ?

Reply:
Done!

Reply:
Hey! I was wondering if having the type in other file will be counter-intuitive for the copy-paste logic of the project